SH4RE/SH4RE/CreateListing/CreateListingView.swift

```diff
@@ -210,57 +220,112 @@ struct CreateListingView: View {
-            }
-            let listingFields = ["Title": title, "Description" : description, "Price" : cost, "Category" : categorySelection, "Availability": calAvail, "Address": postalCode, "UID": getCurrentUserUid()] as [String : Any]
-            let documentID = documentWrite(collectionPath: "Listings", data: listingFields)
-            
-            // upload images and add paths to data fields
-            var index = 1
-            var imgPath = ""
-            var arrayImgs:[String] = []
-            for pic in pictures {
-                imgPath = "listingimages/" + documentID + "/" + String(index) + ".jpg"
-                arrayImgs.append(imgPath)
-                storageManager.upload(image: pic, path: imgPath)
-                index += 1
-            }
-            if (documentUpdate(collectionPath: "Listings", documentID: documentID, data: ["image_path" : arrayImgs])) {
-                NSLog("error");
-            }
-            showPostAlertX = true
-
-            //reset inputs
-            resetInputs()
-        }
+        return true
     }
     
     var body: some View {
         ZStack {
             Color.backgroundGrey.ignoresSafeArea()
+                .onAppear() {
+                    isEditing = editListing.title != ""
+                    if (isEditing) {
+                        title = editListing.title
+                        description = editListing.description
+                        cost = editListing.price
+                        postalCode = editListing.address
+                        availabilityCalendar.selectedDates = editListing.availability
+                        imagesCount = editListing.imagepath.count
+                        categorySelection = editListing.category
+                        imagesCount = (imagesCount <  5) ? imagesCount + 1 : imagesCount
+                        for path in editListing.imagepath {
+                            let storageRef = Storage.storage().reference(withPath: path)
+                            storageRef.getData(maxSize: 1 * 1024 * 1024) { [self] data, error in
+                                if let error = error {
+                                    print (error)
+                                } else {
+                                    //Image Returned Successfully:
+                                    let image = UIImage(data: data!)
+                                    pictures.append(image!)
+                                }
+                            }
+                        }
+                    }
+                }
             
             if (currentUser.isGuest()) {
                 GuestView(tabSelection: $tabSelection).environmentObject(currentUser)
             }
             else {
                 ScrollViewReader { value in
                     ScrollView([.vertical]) {
-                        Text("New Post")
+                        Text(isEditing ? "Edit Listing" : "New Post")
```

Comment:
please make these titles have the same positioning and styling as the home and search pages. Messaging will need to be updated too (separately)

Reply:
can you be more specific, is there a button on the home and search page that has the same style?

Reply:
updated